controllers/spacerequest/spacerequest_controller.go

```diff
@@ -43,12 +63,440 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, memberClusters map[strin
+		reflect.DeepEqual(spaceRequest.Spec.TargetClusterRoles, subSpace.Spec.TargetClusterRoles) {
+		// everything is up to date let's return
+		return false, nil
+	}
+
+	// update tier name and target cluster roles
+	logger.Info("updating subSpace", "name", subSpace.Name)
+	subSpace.Spec.TierName = spaceRequest.Spec.TierName
+	subSpace.Spec.TargetClusterRoles = spaceRequest.Spec.TargetClusterRoles
+	err := r.Client.Update(context.TODO(), subSpace)
+	if err != nil {
+		return false, errs.Wrap(err, "unable to update tiername and targetclusterroles")
+	}
+
+	logger.Info("subSpace updated", "name", subSpace.Name, "target_cluster_roles", subSpace.Spec.TargetClusterRoles, "tierName", subSpace.Spec.TierName)
+	return true, nil
+}
+
+// getParentSpace retrieves the name of the space that provisioned the namespace in which the spacerequest was issued.
+// we need to find the parentSpace so we can `link` the 2 spaces and inherit the spacebindings from the parent.
+func (r *Reconciler) getParentSpace(memberCluster cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest) (*toolchainv1alpha1.Space, error) {
+	parentSpace := &toolchainv1alpha1.Space{}
+	namespace := &v1.Namespace{}
+	err := memberCluster.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: "",
+		Name:      spaceRequest.Namespace,
+	}, namespace)
+	if err != nil {
+		return parentSpace, errs.Wrap(err, "unable to get namespace")
+	}
+	// get owner name which is equal to NSTemplateSet name and Space name
+	parentSpaceName, found := namespace.Labels[toolchainv1alpha1.OwnerLabelKey]
+	if !found || parentSpaceName == "" {
+		err := errs.Errorf("unable to find owner label %s on namespace %s", toolchainv1alpha1.OwnerLabelKey, namespace.GetName())
+		return parentSpace, err
+	}
+
+	// check that parentSpace still exists and is not being deleted
+	err = r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: r.Namespace,
+		Name:      parentSpaceName,
+	}, parentSpace)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// parentSpace was deleted
+			return parentSpace, errs.Wrap(err, "the parentSpace does not exist")
+		}
+		return parentSpace, errs.Wrap(err, "unable to get parentSpace")
+	}
+
+	return parentSpace, nil // all good
+}
+
+func newSpace(spaceRequest *toolchainv1alpha1.SpaceRequest, parentSpaceName, subSpaceNamespace string) *toolchainv1alpha1.Space {
+	labels := map[string]string{
+		toolchainv1alpha1.SpaceRequestLabelKey:          spaceRequest.GetName(),
+		toolchainv1alpha1.SpaceRequestNamespaceLabelKey: spaceRequest.GetNamespace(),
+	}
+
+	space := &toolchainv1alpha1.Space{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    subSpaceNamespace,
+			GenerateName: parentSpaceName + "-",
+			Labels:       labels,
+		},
+		Spec: toolchainv1alpha1.SpaceSpec{
+			TargetClusterRoles: spaceRequest.Spec.TargetClusterRoles,
+			TierName:           spaceRequest.Spec.TierName,
+			ParentSpace:        parentSpaceName,
+		},
+	}
+	return space
+}
```

Comment:
there is already https://github.com/codeready-toolchain/host-operator/blob/master/controllers/usersignup/space.go so you can make it more generic, move it to pkg folder, and use it also here.
if making it generic is not worth it - because the input values are just too different, then at least put both of the functions into one file.

Reply:
I added a new `pkg/space` in [60159bc](https://github.com/codeready-toolchain/host-operator/pull/770/commits/60159bc49fc4872acb0b880853cb3775e5454300).

For now I've just added the two functions, because making something more flexible will look very similar to what we have in https://github.com/codeready-toolchain/host-operator/blob/master/test/space/space.go or https://github.com/codeready-toolchain/toolchain-e2e/blob/master/testsupport/space.go

Creating something that can be used both in tests and prod code I think will end up in a bigger refactor which IMO will create some confusion in this PR.

But probably is something we can consider for the future, I see many similar space related functions that are now used for unit and e2e tests which we could refactor in a pkg maybe in `toolchain-common` repo and the reuse those also in some prod code. Something to think about for the future maybe.

Reply:
Totally fine as you did in the commit :+1: Thanks